pkmntypes.py

```diff
@@ -22,3 +22,28 @@ def __init__(self, **kwargs):
 
 class Pokemon():
 	pass
+
+class BattleContext():
+	"""This is a class that describes the point of view of a given Pokemon on the battlefield. It provides
+	enough information for a user to make an informed decision about what turn to make next.   
+	"""
+	# will add make the docstrings comply with the sphinx format once this class is looking good
+
+	def __init__(self):
+		self.battle = battle
+		self.pokemon = pokemon
+		self.team = team
+		self.targets = targets
+		self.allies = allies
+		self.opponents = opponents
+
+	def parseJSON(self) -> dict:
```

Comment:
Would rather have usage be like this:
```python
data = json.loads(text)
context = BattleContext(**data)
```

Reply:
Would `context` be initialized inside the `BattleContext` class?

Reply:
No, `context` in this sample _is_ an instance of `BattleContext`. This is meant to be an example of how you would use the class in practice.